File: rust/src/models/slot_fee.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct SlotFee {
    pub initial_slots: i32,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub max_slots: Option<i32>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub min_slots: Option<i32>,

    pub unit: String,

    pub unit_rate: String,
}

impl SlotFee {
    pub fn new(initial_slots: i32, unit: String, unit_rate: String) -> Self {
        Self {
            initial_slots,
            max_slots: None,
            min_slots: None,
            unit,
            unit_rate,
        }
    }
}

File: typescript/src/models/slotFee.ts
// this file is @generated

export interface SlotFee {
  initialSlots: number;

  maxSlots?: number | null;

  minSlots?: number | null;

  unit: string;

  unitRate: string;
}

export const SlotFeeSerializer = {
  _fromJsonObject(object: any): SlotFee {
    return {
      initialSlots: object["initial_slots"],
      maxSlots: object["max_slots"],
      minSlots: object["min_slots"],
      unit: object["unit"],
      unitRate: object["unit_rate"],
    };
  },

  _toJsonObject(self: SlotFee): any {
    return {
      initial_slots: self.initialSlots,
      max_slots: self.maxSlots,
      min_slots: self.minSlots,
      unit: self.unit,
      unit_rate: self.unitRate,
    };
  },
};
